installer/internal/algo/ubuntu22_04k8s.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package algo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+)
+
+// Ubuntu22_04Installer represent the installer implementation for ubuntu22.04.* os distribution
+type Ubuntu22_04Installer struct {
+	install   string
+	uninstall string
+}
+
+// NewUbuntu22_04Installer will return new Ubuntu22_04Installer instance
+func NewUbuntu22_04Installer(ctx context.Context, arch, bundleAddrs string) (*Ubuntu22_04Installer, error) {
+	parseFn := func(script string) (string, error) {
+		parser, err := template.New("parser").Parse(script)
+		if err != nil {
+			return "", fmt.Errorf("unable to parse install script")
+		}
+		var tpl bytes.Buffer
+		if err = parser.Execute(&tpl, map[string]string{
+			"BundleAddrs":        bundleAddrs,
+			"Arch":               arch,
+			"ImgpkgVersion":      ImgpkgVersion,
+			"BundleDownloadPath": "{{.BundleDownloadPath}}",
+		}); err != nil {
+			return "", fmt.Errorf("unable to apply install parsed template to the data object")
+		}
+		return tpl.String(), nil
+	}
+
+	install, err := parseFn(DoUbuntu22_04K8s)
+	if err != nil {
+		return nil, err
+	}
+	uninstall, err := parseFn(UndoUbuntu22_04K8s)
+	if err != nil {
+		return nil, err
+	}
+	return &Ubuntu22_04Installer{
+		install:   install,
+		uninstall: uninstall,
+	}, nil
+}
+
+// Install will return k8s install script
+func (s *Ubuntu22_04Installer) Install() string {
+	return s.install
+}
+
+// Uninstall will return k8s uninstall script
+func (s *Ubuntu22_04Installer) Uninstall() string {
+	return s.uninstall
+}
+
+// contains the installation and uninstallation steps for Ubuntu 22.04 and k8s
+var (
+	DoUbuntu22_04K8s = `
```

Comment:
Can we separate out this shell script template into separate sh files? 
It's difficult to `diff` from here if the templates for Ubuntu 22 and Ubuntu 22 are identical or not - but there's a lot of duplication.